packages/api/src/services/lsp-state.ts

```diff
@@ -0,0 +1,127 @@
+import * as uma from "@uma/sdk";
+import { BatchRead, toString, toNumber, parseBytes, nowS } from "../libs/utils";
+import { AppState } from "..";
+import { lsps } from "../tables";
+
+type Instance = uma.clients.lsp.Instance;
+type Config = undefined;
+type Dependencies = Pick<
+  AppState,
+  "lsps" | "registeredLsps" | "provider" | "collateralAddresses" | "shortAddresses" | "longAddresses" | "multicall"
+>;
+
+export default (config: Config, appState: Dependencies) => {
+  const { lsps, registeredLsps, provider, collateralAddresses, shortAddresses, longAddresses, multicall } = appState;
+
+  // default props we want to query on contract
+  const staticProps: [string, (x: any) => any][] = [
+    ["collateralPerPair", toString],
+    ["priceIdentifier", parseBytes],
+    ["collateralToken", toString],
+    ["longToken", toString],
+    ["shortToken", toString],
+    ["finder", toString],
+    ["financialProductLibrary", toString],
+    ["customAncillaryData", toString],
+    ["prepaidProposerReward", toString],
+  ];
+
+  const dynamicProps: [string, (x: any) => any][] = [
+    ["expiryPrice", toString],
+    ["expiryPercentLong", toString],
+    ["expirationTimestamp", toNumber],
+    ["contractState", toNumber],
+  ];
```

Comment:
this is an optimization. we define the properties we consider static and dynamic, so that we only need to call static functions the first time we see a new contract, where dynamic props will get called every time update runs.